### PR TITLE
Added Sign-up Flow in ES Language

### PR DESCRIPTION
**Pull request by faezeh-deriv**
# Summary
This PR is for adding the sign-up flow in `ES` language after [WEBREL-2604](https://app.clickup.com/t/20696747/WEBREL-2604) production bug. The test is running on mobile view.

# Ticket
https://app.clickup.com/t/20696747/TRAH-3089

# Note
N/A

### CODE REVIEW

File: cypress/e2e/wip/Tradershub/accountSignupES.cy.js [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+  })
+  cy.findByTestId('last_name').type('automation acc')
+  // interacting with date picker in desktop mode
+  cy.viewport('macbook-16')
+  cy.findByTestId('date_of_birth').click()
+  cy.findByText('2006').click()
+  cy.findByText('feb.').click()
+  cy.findByText('9', { exact: true }).click()
+  cy.viewport('iphone-xr')
+  cy.get('select[data-testid="place_of_birth_mobile"]').select(country)
```
Comment (meeyun-deriv) [inline cypress/e2e/wip/Tradershub/accountSignupES.cy.js:78]: please use testing library whenever possible. E.g. 
```
    cy.findByTestId('place_of_birth').click();
```

File: cypress/e2e/wip/Tradershub/accountSignupES.cy.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  cy.findByTestId('phone').type('12345678')
+  cy.get('select[data-testid="citizenship_mobile"]').select(country)
```
Comment (meeyun-deriv) [inline cypress/e2e/wip/Tradershub/accountSignupES.cy.js:80]: same for comment above. Please check across the PR.

**Review by meeyun-deriv — CHANGES_REQUESTED** [review]
Many custom command could have been duplicated with existing ones from create account. Since we are adding translation check for create account, we should update the existing custom command to support multi languages instead

File: cypress/e2e/wip/Tradershub/accountSignupES.cy.js (hunk 1 above)
Comment (faezeh-deriv) [reply cypress/e2e/wip/Tradershub/accountSignupES.cy.js:78]: I tried this but since it was the mobile view, it didn't work - But let me try it again.

**Comment by faezeh-deriv** [comment]
> Many custom command could have been duplicated with existing ones from create account. Since we are adding translation check for create account, we should update the existing custom command to support multi languages instead

After all considerations, I don't think making the sign-up commands that general will be a good idea:
1. It will add to the command's complexity
2. There will be too many `if else` in one command and the runtime will increase
3. In my case, I am running the test in mobile and the already defined commands are compatible with desktop (element locators and some commands used are different)
4. Again, in my case, the URLs and redirections (the scenario) are also different in some places
In conclusion, I don't suggest making the commands general as it is not an efficient thing to do.

Let me know what do you think please. Thank you.

But if we insist on making the commands general, I can make commands for other languages rather than `EN` and those commands can be used in any language. So, we will have 2 sets of commands for sign-up:
1. Sign-up commands in `EN`
2. Sign-up commands in any other language rather than `EN`

**Comment by faezeh-deriv** [comment]
Update: I talked with @abuzar-deriv and @bilalmajid-deriv, we came to a conclusion to create a fixture `.json` file and define different language translations in it so we can pass them to the same commands across the sign-up flow test cases (instead of creating different commands for different languages)

But we will need to create different sets of commands for mobile and desktop if the steps duplication is not a lot.

File: cypress/e2e/wip/Tradershub/accountSignupES.cy.js [hunk 3]
```diff
@@ -0,0 +1,217 @@
+import '@testing-library/cypress/add-commands'
+import { generateEpoch } from '../../../support/helper/utility'
+
+Cypress.Commands.add('c_setEndpointES', (signUpMail) => {
+  localStorage.setItem('config.server_url', Cypress.env('stdConfigServer'))
+  localStorage.setItem('config.app_id', Cypress.env('stdConfigAppId'))
+  const mainURL = 'https://staging-app.deriv.com/endpoint?lang=ES'
```
[the remark this answers is not in the document]
Comment (faezeh-deriv) [reply cypress/e2e/wip/Tradershub/accountSignupES.cy.js]: the hardcode URL is removed.

File: cypress/e2e/wip/Tradershub/accountSignupES.cy.js (hunk 1 above)
Comment (faezeh-deriv) [reply cypress/e2e/wip/Tradershub/accountSignupES.cy.js]: this is done.

**Comment by faezeh-deriv** [comment]
> Many custom command could have been duplicated with existing ones from create account. Since we are adding translation check for create account, we should update the existing custom command to support multi languages instead

this is done using fixtures.

**Review by abuzar-deriv — CHANGES_REQUESTED** [review]
@faezeh-deriv , it's breaking some English language tests when run right after Spanish test. You might need to clear some cookier and/or local storage in commands where it's visiting a URL. 
Can you please also run a few other signup tests just to check nothing else is breaking?
Thanks.

File: cypress/support/commands/tradersHub.js [hunk 4]
```diff
@@ -132,106 +179,135 @@ Cypress.Commands.add(
     options = {}
   ) => {
     const { isMobile = false } = options
```
[the remark this answers is not in the document]
Comment (faezeh-deriv) [reply cypress/support/commands/tradersHub.js:181]: done.

File: cypress/support/commands/tradersHub.js [hunk 5]
```diff
@@ -253,72 +329,87 @@ Cypress.Commands.add('c_addAccount', () => {
...
-    'be.visible'
-  )
-  cy.findByRole('button', { name: 'Next' }).should('be.disabled')
-  if (isMobile) cy.get(`select[name='country_input']`).select(CoR)
-  else {
-    cy.findByLabelText('Country').type(CoR)
-    cy.findByText(CoR).click()
-  }
-  cy.findByRole('button', { name: 'Next' }).should('not.be.disabled')
+  const { language = 'english' } = options
```
[the remark this answers is not in the document]
Comment (faezeh-deriv) [reply cypress/support/commands/tradersHub.js:332]: done.

File: cypress/support/commands/tradersHub.js [hunk 6]
```diff
@@ -253,72 +329,87 @@ Cypress.Commands.add('c_addAccount', () => {
...
-  let count = 1
-  while (count < 5) {
-    if (isMobile) {
-      cy.get('.dc-select-native__picker')
-        .eq(count - 1)
-        .select(2)
-    } else {
-      cy.findAllByTestId('dt_dropdown_display').eq(count).click()
-      cy.findAllByTestId('dti_list_item').eq(2).click()
+  const { language = 'english' } = options
```
[the remark this answers is not in the document]
Comment (faezeh-deriv) [reply cypress/support/commands/tradersHub.js:361]: done.

File: cypress/support/commands/tradersHub.js [hunk 7]
```diff
@@ -253,72 +329,87 @@ Cypress.Commands.add('c_addAccount', () => {
...
-  while (count < 9) {
-    if (isMobile) {
-      cy.get('.dc-select-native__picker')
-        .eq(count - 1)
-        .select(1)
-    } else {
-      cy.findAllByTestId('dt_dropdown_display').eq(count).click()
-      cy.findAllByTestId('dti_list_item').eq(1).click()
-    }
+  const { language = 'english' } = options
```
[the remark this answers is not in the document]
Comment (faezeh-deriv) [reply cypress/support/commands/tradersHub.js:395]: done.

File: cypress/support/commands/tradersHub.js [hunk 8]
```diff
@@ -328,69 +419,128 @@ Cypress.Commands.add('c_completeFatcaDeclarationAgreement', () => {
...
-  cy.findByRole('heading', { name: 'Account added' }).should('be.visible')
-  cy.findByText(
-    'Your account will be available for trading once the verification of your account is complete.'
-  ).should('be.visible')
-  cy.findByRole('button', { name: 'Verify now' })
-    .should('be.visible')
-    .and('be.enabled')
-  cy.findByRole('button', { name: 'Maybe later' }).click()
-  cy.c_completeTradersHubTour()
+  const { language = 'english' } = options
```
[the remark this answers is not in the document]
Comment (faezeh-deriv) [reply cypress/support/commands/tradersHub.js:422]: done.

**Comment by faezeh-deriv** [comment]
> @faezeh-deriv , it's breaking some English language tests when run right after Spanish test. You might need to clear some cookier and/or local storage in commands where it's visiting a URL. Can you please also run a few other signup tests just to check nothing else is breaking? Thanks.

this is done.

**Review by abuzar-deriv — APPROVED** [review]
LGTM

**Review by markodonnell-deriv — APPROVED** [review]
Looks good
